Add more ads plc types

philipperequile opened the pull request:
Support for 2 new 4-byte PLC datatypes DINT and UDINT

## Description:


**Related issue (if applicable):** fixes #<home-assistant issue number goes here>

**Pull request in [home-assistant.io](https://github.com/home-assistant/home-assistant.io) with documentation (if applicable):**
home-assistant/home-assistant.io#8700

## Example entry for `configuration.yaml` (if applicable):
```yaml
sensor:
  - platform: ads
    adsvar: Energy.HAEnergyTot
    adstype: udint  
    unit_of_measurement: 'kWh'
    factor: 1000
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [x] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [ ] New or updated dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [x] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

philipperequile committed:
Update __init__.py

Support for 2 new 4-byte PLC datatypes DINT and UDINT

homeassistant commented:
Hi @philipperequile, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#19801).

Once you do that we will be able to review and accept this pull request.

Thanks!

philipperequile committed:
Update __init__.py

fabaff committed:
Fix lint issue

philipperequile committed:
Merge branch 'dev' into patch-1

philipperequile committed:
Update __init__.py

MartinHjelmare commented on homeassistant/components/ads/__init__.py:
```diff
@@ -46,7 +52,8 @@
 SCHEMA_SERVICE_WRITE_DATA_BY_NAME = vol.Schema({
     vol.Required(CONF_ADS_TYPE):
-        vol.In([ADSTYPE_INT, ADSTYPE_UINT, ADSTYPE_BYTE, ADSTYPE_BOOL]),
+        vol.In([ADSTYPE_INT, ADSTYPE_UINT, ADSTYPE_BYTE, ADSTYPE_BOOL, /
```
Break the line after the bracket instead. Make sure to end each bracket or parenthesis with the corresponding bracket or parenthesis.

MartinHjelmare commented on homeassistant/components/ads/__init__.py:
```diff
@@ -21,21 +23,25 @@
 
 SERVICE_WRITE_DATA_BY_NAME = 'write_data_by_name'
 
+TYPES = [ADSTYPE_INT, ADSTYPE_UINT, ADSTYPE_BYTE, ADSTYPE_DINT, ADSTYPE_UDINT]
```
This isn't used.

MartinHjelmare replied:
Is this used anywhere?

MartinHjelmare replied:
Can we remove this?

philipperequile replied:
I don't think I inserted this type. I removed it in my test environment, and de beckhoff ads seneors are working fine.

gijbelsy commented:
Is it possible to add REAL values too?

philipperequile committed:
Merge branch 'dev' into patch-1

balloob commented:
@philipperequile you can't just mark a conversation to make the issues go away ;-) Hound was right, it's invalid syntax. Did you test it locally?

philipperequile committed:
Update __init__.py

philipperequile committed:
Update __init__.py

philipperequile commented:
> @philipperequile you can't just mark a conversation to make the issues go away ;-) Hound was right, it's invalid syntax. Did you test it locally?
Sorry @balloob , I'm new to github and python. I corrected and tested in my test environment.

philipperequile committed:
Support for 2 new PLC types

Sensor supports DINT and UDINT PLC types

philipperequile commented:
Hi

I removed this line and tested. No problems found.

Op do 21 feb. 2019 om 23:24 schreef Martin Hjelmare <
notifications@github.com>:

> *@MartinHjelmare* commented on this pull request.
> ------------------------------
>
> In homeassistant/components/ads/__init__.py
> <https://github.com/home-assistant/home-assistant/pull/19801#discussion_r259142284>
> :
>
> >
>  SERVICE_WRITE_DATA_BY_NAME = 'write_data_by_name'
>
> +TYPES = [ADSTYPE_INT, ADSTYPE_UINT, ADSTYPE_BYTE, ADSTYPE_DINT, ADSTYPE_UDINT]
>
> Is this used anywhere?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/home-assistant/pull/19801#discussion_r259142284>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AQttznwMUym27oO25uDldSfIQHuCAGsEks5vPxysgaJpZM4ZyCM9>
> .
>

philipperequile committed:
Update __init__.py

I removed unused TYPES = [ADSTYPE_INT, ADSTYPE_UINT, ADSTYPE_BYTE, ADSTYPE_DINT, ADSTYPE_UDINT]

MartinHjelmare reviewed (approved):
Thanks!

MartinHjelmare commented:
Can be merged when build passes and a docs PR is linked in the PR description.

philipperequile commented:
Documentation is updated.
https://github.com/home-assistant/home-assistant.io/pull/8700
